### PR TITLE
rate_limit: statusor to avoid potential template argument deduction

### PR DESCRIPTION
**Pull request by botengyao**
Commit Message:
Additional Description:
Risk Level: low
Testing: 
Docs Changes:
Release Notes:

### CODE REVIEW

File: source/extensions/filters/http/rate_limit_quota/filter.cc [hunk 1]
```diff
@@ -30,7 +30,8 @@ Http::FilterHeadersStatus RateLimitQuotaFilter::decodeHeaders(Http::RequestHeade
   // succeeds.
   const RateLimitOnMatchAction& match_action =
       match_result.value()->getTyped<RateLimitOnMatchAction>();
-  absl::StatusOr ret = match_action.generateBucketId(*data_ptr_, factory_context_, visitor_);
+  absl::StatusOr<BucketId> ret =
```
Comment (tyxia) [inline source/extensions/filters/http/rate_limit_quota/filter.cc:33]: cc @bsurber who initially made the change.

**Review by tyxia — APPROVED** [review]

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

**Pull request merged**